src/robot.coffee

```diff
@@ -247,6 +264,28 @@ class Robot
   # Returns nothing.
   # Returns before executing callback
   receive: (message, cb) ->
+    # When everything is finished (down the middleware stack and back up),
+    # pass control back to the robot
+    receiveMiddlewareDone = () ->
+      if cb?
+        process.nextTick -> cb true
+
+    @middleware.receive.execute(
+      {response: new Response(this, message)}
```

Comment:
Just saw the example from below. Yeah, I really think listener middleware is the correct place for that kind of blacklisting. While it could be accomplished using receive middleware, it should be considered a hack (though the whole point of middleware is to allow this kind of customization).

Drawing parallels to Express: in Express, you have a single request/response pair. When a request comes in, a response object is generated to handle responding to that request. In Hubot, we have multiple "requests": one request per matched listener. As a result, there is one response object for handling the response from that listener.

We could change the model so that a message is considered a request and there is a single response object shared across all matching listeners. Alternatively, we could decide that the Express request/response model doesn't work well for us and define something completely different.